src/App.jsx

```diff
@@ -14,33 +17,117 @@ export const goodsFromServer = [
+const NO = 'no';
+const INITIAL = 'initial';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, { sortField, reverse }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        case INITIAL:
+        default:
+          return 0;
+      }
+    });
+
+    if (reverse === YES) {
+      return preparedGoods.sort(() => -1);
+    }
+
+    if (reverse === NO) {
+      return preparedGoods.sort(() => 0);
+    }
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState(INITIAL);
+  const [reverse, setReverse] = useState(NO);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverse,
+  });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_NAME,
+          })}
+          onClick={() => {
+            setSortField(SORT_FIELD_NAME);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => {
+            setSortField(SORT_FIELD_LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': reverse !== YES,
+          })}
+          onClick={() => {
+            if (reverse === NO) {
+              setReverse(YES);
+            }
+
+            if (reverse === YES) {
+              setReverse(NO);
+            }
+          }}
+        >
+          Reverse
+        </button>
+        {(sortField === INITIAL && reverse === YES) || sortField !== INITIAL ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(INITIAL);
+              setReverse(NO);
+            }}
+          >
+            Reset
+          </button>
+        ) : (
+          ''
+        )}
```

Comment:
Hello, @arthurfujii.

In react, we can use another approach to render components conditionally.

You can try something like that.

```js
{((sortField === INITIAL && reverse === YES) || sortField !== INITIAL) && 
    <button
         type="button"
         className="button is-danger is-light"
         onClick={() => {
            setSortField(INITIAL);
            setReverse(NO);
          }}
    >
        Reset
   </button>
}
```

With that, you don't need the if/else statement and render a empty string if it be false.

Can you please try that?